apiservers/portlayer/restapi/middleware/attach_hijacker.go

```diff
@@ -0,0 +1,193 @@
+// Copyright 2016 VMware, Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package middleware
+
+import (
+	//	"bufio"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"net/url"
+	"strings"
+	"time"
+
+	"golang.org/x/net/context"
+
+	log "github.com/Sirupsen/logrus"
+
+	"github.com/docker/docker/pkg/stdcopy"
+
+	"github.com/vmware/vic/apiservers/portlayer/models"
+	"github.com/vmware/vic/pkg/attachutils"
+	"github.com/vmware/vic/portlayer/attach"
+)
+
+type AttachHijacker struct {
+	origHandler  http.Handler
+	attachServer *attach.Server
+
+	containerID string
+	useTty      bool
+	useStdin    bool
+	useStdout   bool
+	useStderr   bool
+	stream      bool
+}
+
+const basePath = "/containers"
+const operation = "/attach"
+
+//NewAttachHijacker handles attach for the interaction handler.  With the current
+//version of go-swagger, we cannot implement attach with a standard swagger
+//handler.
+func NewAttachHijacker(orig http.Handler, as *attach.Server) *AttachHijacker {
+	return &AttachHijacker{origHandler: orig, attachServer: as}
+}
+
+func (a *AttachHijacker) ServeHTTP(rw http.ResponseWriter, r *http.Request) {
+	// Make sure the call is for /containers/{id}/attach
+	if !strings.HasPrefix(r.URL.Path, basePath) || !strings.Contains(r.URL.Path, operation) {
+		a.origHandler.ServeHTTP(rw, r)
+		return
+	}
+
+	code, err := a.GetParams(r)
+	if err != nil {
+		log.Errorf("interaction attach container failed %#v\n", err)
+		a.WritePortlayerError(rw, code, err.Error())
+		return
+	}
+
+	getStream := attachutils.HijackConnection(rw, r)
```

Comment:
I also thought of doing that.  I left it as is so we have more flexibility when we upgrade our Swagger version.  It would provide us with a mechanism to get at these streams in functions that don't have direct access to the http responsewriter.  However, I do see the point of keeping it simple now and have no issue changing it.